airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -3,37 +3,59 @@
 #
 
 import asyncio
+import itertools
 import logging
-from datetime import datetime
 from functools import cache
-from typing import Any, Iterable, List, Mapping, Optional, Union
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Union
 
-from airbyte_cdk.models import SyncMode
-from airbyte_cdk.sources.declarative.types import StreamSlice, StreamState
 from airbyte_cdk.sources.file_based.exceptions import MissingSchemaError, RecordParseError, SchemaInferenceError
 from airbyte_cdk.sources.file_based.remote_file import RemoteFile
-from airbyte_cdk.sources.file_based.schema_helpers import merge_schemas, type_mapping_to_jsonschema
+from airbyte_cdk.sources.file_based.schema_helpers import merge_schemas
 from airbyte_cdk.sources.file_based.schema_validation_policies import record_passes_validation_policy
 from airbyte_cdk.sources.file_based.stream import AbstractFileBasedStream
+from airbyte_cdk.sources.file_based.stream.cursor import FileBasedCursor
+from airbyte_cdk.sources.file_based.types import StreamSlice
+from airbyte_cdk.sources.streams import IncrementalMixin
 from airbyte_cdk.sources.utils.record_helper import stream_data_to_airbyte_message
 
 
-class DefaultFileBasedStream(AbstractFileBasedStream):
+class DefaultFileBasedStream(AbstractFileBasedStream, IncrementalMixin):
```

Comment:
I lean towards a unified flow for simplicity. Trying to optimize large full refresh syncs feels like a goose chase because we can't checkpoint.

I can take a stab at splitting the flow, but I suspect we'll still need to use the same class because the [FileBasedSource doesn't know about the sync_mode when instantiating the stream](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_source.py#L80).

Reply:
Good point re the simplicity. Let's leave it as-is for now. If we need to do an optimization later we always can.